### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by tim-field** [openlayers/ol-mapbox-style#269]
Hi it seems this library doesn't have any type definitions, is that correct or am I missing something?   It does have this file `tsconfig-typecheck.json` 😕

**Comment by ahocevar** [comment]
We use Typescript for type checking the library. This is possible thanks to JSDoc type annotations we. You can also take advantage of those, by configuring Typescript in your project in a way like described in the [OpenLayers README](https://github.com/openlayers/openlayers#intellisense-support-and-type-checking-for-vs-code).

**Comment by ron4stoppable** [comment]
@tim-field, as @ahocevar there is already JSDoc type annotation added to this library and tsconfig can be updated to use the same, you can update `tsconfig.json`  like


```json
{
  "compilerOptions": {
    "allowJs": true,
    "baseUrl": "./",
    "paths": {
      "ol-mapbox-style": ["node_modules/ol-mapbox-style/src"],
      "ol-mapbox-style/*": ["node_modules/ol-mapbox-style/src/*"]
    }
  },
  "include": [
    "**/*.ts",
    "node_modules/ol-mapbox-style/**/*"
  ],
  "typeAcquisition": {
    "exclude": ["ol-mapbox-style"]
  }
}
```

**Comment by ca0v** [comment]
I cannot figure out how to use this module from typescript.  I've been happily coding along using snowpack but now that I want to add support for vector tiles, I am seeing this error:

[snowpack] Package "@mapbox/mapbox-gl-style-spec" exists but package.json "exports" does not include entry for "./dist/index.es.js".

Seems to be referring to this:

```
 "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.es.js"
    },
  },
```

See [exports](https://nodejs.org/api/packages.html#packages_exports).

**Comment by ahocevar** [comment]
Are you sure this is an issue with `ol-mapbox-style`? The error message you posted above refers to `@mapbox/mapbox-gl-style-spec`, which is one of `ol-mapbox-style`'s dependencies. Try if the error goes away when you add `"skipLibCheck": true` to the `"compilerOptions"` of your `tsconfig.json`.

**Comment by ca0v** [comment]
That resolved the typescript issue but `snowpack` is usable to make sense of the `mapbox-gl-style-spec` package.  I'm not understanding what snowpack is complaining about.  I figured since skypack contained these libraries I would just work.

My small repo is [here](https://github.com/ca0v/ol3-ags/tree/vectortile) with the following package.json deps:

```
  "devDependencies": {
    "@snowpack/plugin-typescript": "^1.2.1",
    "reverse-proxy-cache": "https://github.com/ca0v/reverse-proxy-cache/tarball/v1.0.4",
    "snowpack": "^3.8.8"
  },
  "dependencies": {
    "@arcgis/core": "^4.20.2",
    "ol": "^6.6.1",
    "typescript": "^4.4.2"
  },
```

When I run `npx snowpack dev` I get that error:

`[24:27:26] [snowpack] Package "@mapbox/mapbox-gl-style-spec" exists but package.json "exports" does not include entry for "./dist/index.es.js".
`

So I guess I am actually looking for Snowpack Support now.

**Comment by ahocevar** [comment]
Thanks for sharing your findings, @ca0v. I just published v6.4.2 with the fix you linked. Can you please try if it works now?

**Comment by easbar** [comment]
Is this still the recommended way to enable typescript support: https://github.com/openlayers/ol-mapbox-style/issues/269#issuecomment-627263528? Is there any way to use typescript without setting `allowJs: true`?

I'm asking because starting from OpenLayers 6.6.0 the OL types are included in OL again: [6.6.0](https://github.com/openlayers/openlayers/releases/tag/v6.6.0)

**Comment by ahocevar** [comment]
We could add auto-generated types to this library too, then you won't need any special settings in your `tsconfig.json`. I'm going to prepare that for the next release.
